crates/astria-proto/src/native/sequencer/ics20_withdrawal.rs

```diff
@@ -0,0 +1,171 @@
+use std::{
+    error::Error,
+    fmt::Display,
+    str::FromStr,
+};
+
+use ibc_types::{
+    core::{
+        channel::ChannelId,
+        client::Height as IbcHeight,
+    },
+    IdentifierError,
+};
+use penumbra_proto::penumbra::core::component::ibc::v1alpha1::FungibleTokenPacketData;
+
+use super::{
+    asset::{
+        IbcAsset,
+        IbcAssetError,
+    },
+    v1alpha1::IncorrectAddressLength,
+};
+use crate::{
+    generated::{
+        primitive::v1::Uint128,
+        sequencer::v1alpha1 as raw,
+    },
+    native::sequencer::v1alpha1::Address,
+};
+
+#[derive(Debug, Clone)]
+pub struct Ics20Withdrawal {
```

Comment:
I increasingly lean toward abstaining from making all of these private by default.

Can you lock it down?

Reply:
you mean make the fields private? yeah i changed it, made getters instead

Reply:
Yeah, that's what I meant! Sorry, I meant to write `public` instead of `private`.